prusti-contracts/prusti-specs/src/lib.rs

```diff
@@ -659,21 +748,196 @@ pub fn predicate(tokens: TokenStream) -> TokenStream {
     parsed.into_token_stream()
 }
 
-pub fn type_model(attr: TokenStream, tokens: TokenStream) -> TokenStream {
-    let item: syn::Item = handle_result!(syn::parse2(tokens));
+pub fn rewrite_prusti_attributes_for_types(
+    outer_attr_kind: SpecAttributeKind,
+    outer_attr_tokens: TokenStream,
+    item_tokens: TokenStream,
+) -> TokenStream {
+    let mut item: syn::DeriveInput = handle_result!(syn::parse2(item_tokens));
 
-    match item {
+    // Start with the outer attribute
+    let mut prusti_attributes = vec![(outer_attr_kind, outer_attr_tokens)];
+
+    // Collect the remaining Prusti attributes, removing them from `item`.
+    prusti_attributes.extend(extract_prusti_attributes_for_types(&mut item));
+
+    if prusti_attributes.len() > 1 && prusti_attributes
+        .iter()
+        .any(|(ak, _)| ak == &SpecAttributeKind::Trusted)
+    {
+        return syn::Error::new(
+            item.span(),
+            "`trusted!` is incompatible with other Prusti attributes",
+        )
+        .to_compile_error();
+    }
+
+    //we order die attributes to ensure a model attribute is processed first
```

Comment:
the*

Also is this relying on the fact that `model` comes before `print_counterexample` alphabetically? If so, that's very hacky.

Reply:
@mlimbeck

Reply:
done.